### PR TITLE
209 penalty fix

### LINKED ISSUE #209: Penalty weights are incorrect
**Describe the bug**
See title.

**To Reproduce**
1. Any segmentation/classification problem
2. Take a print at the end of `GANDLF.utils.tensor.get_class_imbalance_weights`
3. See that `penalty_dict` and `weights_dict` are the same

For example, for a small I-SPY dataset:
```
penalty_dict: {0: 0.9928165871714076, 1: 0.0071834128285923945} 
weights_dict: {0: 0.9928165871714076, 1: 0.0071834128285923945}
```

**Expected behavior**
They should be `1-each_other`. In the above example, it should be this:

```
penalty_dict: {0: 0.0071834128285923945, 1: 0.9928165871714076} 
weights_dict: {0: 0.9928165871714076,    1: 0.0071834128285923945}
```

**Screenshots**
N.A.

**GaNDLF Version**
0.0.12-dev

**Desktop (please complete the following information):**
N.A.

**Additional context**
Reported by @iskobleva

### PR DESCRIPTION
**Pull request by sarthakpati**
Fixes #209

## Proposed Changes

- penalty calculation fixed
- new output for same example (in issue):
```
penalty_dict: {0: 0.006036875965002797, 1: 0.9939631240349976} 
weights_dict: {0: 0.9939631240349974,   1: 0.006036875965002573}
```
Values are slightly different because of the difference in how the data is split between training and validation.

## Checklist

- [x] I have read the [`CONTRIBUTING`](https://github.com/CBICA/GaNDLF/blob/master/CONTRIBUTING.md) guide
- [x] My PR is based from the [current GaNDLF master ](https://garygregory.wordpress.com/2016/11/10/how-to-catch-up-my-git-fork-to-master/)
- [x] Non-breaking change (would not break existing functionality): if changes breaks current code, please provide as many details as possible
- [x] Function/class source code documentation added/updated
- [x] Code has been [blacked](https://github.com/psf/black#usage) for style consistency
- [x] If applicable, version information [has been updated in GANDLF/version.py](https://github.com/CBICA/GaNDLF/blob/master/GANDLF/version.py)
- [x] If adding a submodule, add to list of exceptions for black styling in [pyproject.toml](https://github.com/CBICA/GaNDLF/blob/master/pyproject.toml) file 
- [x] [Usage documentation](https://github.com/CBICA/GaNDLF/blob/master/docs) has been updated, if appropriate
- [x] [History](https://github.com/CBICA/GaNDLF/blob/master/HISTORY.md) has been updated, if appropriate
- [x] Tests added or modified to [cover the changes](https://app.codecov.io/gh/CBICA/GaNDLF); if coverage is reduced, please give explanation

### CODE REVIEW

File: GANDLF/utils/tensor.py [hunk 1]
```diff
@@ -245,15 +241,12 @@ def get_class_imbalance_weights(training_data_loader, parameters):
                     total_counter += 1
 
     # Normalize class weights
-    weights_dict = {key: val / total_counter for key, val in abs_dict.items()}
-
+    weights_dict = {key: (val + sys.float_info.epsilon) / total_counter for key, val in abs_dict.items()}
+
```
Comment (iskobleva) [inline GANDLF/utils/tensor.py:245]:
```suggestion
    weights_dict = {key: (val + sys.float_info.epsilon) / total_counter for key, val in abs_dict.items()}
    penalty = {key: total_counter / (len(abs_dict) * val) for key, val in abs_dict.items()}
    penalty_sum = np.fromiter(penalty.values(), dtype=np.float64).sum()
    penalty_dict = {key: val / penalty_sum for key, val in penalty.items()} # normalize penalty to sum of 1
```

File: GANDLF/utils/tensor.py [hunk 2]
```diff
@@ -245,15 +241,12 @@ def get_class_imbalance_weights(training_data_loader, parameters):
                     total_counter += 1
 
     # Normalize class weights
-    weights_dict = {key: val / total_counter for key, val in abs_dict.items()}
-
+    weights_dict = {key: (val + sys.float_info.epsilon) / total_counter for key, val in abs_dict.items()}
+    
     # get the penalty values - abs_dict contains the overall number for each class in the penalty data
```
Comment (iskobleva) [inline GANDLF/utils/tensor.py:246]: lines below can be deleted

**Review by Geeks-Sid — APPROVED** [review]

**Comment by Geeks-Sid** [comment]
@sarthakpati  why is the codecov/project failing?

**Comment by sarthakpati** [comment]
> @sarthakpati why is the codecov/project failing?

Well, probably because the number of lines of code reduced, which ensured that the number of lines covered by pytest reduced:

image

**Comment by sarthakpati** [comment]
So is this good to merge, @Geeks-Sid and @iskobleva?

**Comment by sarthakpati** [comment]
@iskobleva confirms that this works for loss calculation.

**Pull request merged**